File: src/c/binsearch/binsearch.c
#include <stdlib.h>

int binsearch(int* array, size_t length, int x)
{
    size_t l = 0;
    size_t r = length-1;

    while(l<=r)
    {
        size_t mid = (l+r)/2;
        if(*(array+mid)==x)
            return 1;
        else if(*(array+mid)<x)
            r = mid - 1;
        else
            l = mid + 1;
    }

    return 0;
}

File: src/js/bin_search.js
// returns true if the element is found
function bin_search(array, searched)
{
    let left = 0;
    let right = array.length-1;

    while(left<right)
    {
        let mid = (right-left)/2 + left;
        if(array[mid]==searched)
            return true;
        else if(array[mid]>searched)
            right = mid - 1;
        else
            left = mid + 1;
    }

    return false;
}